destination/config.go

```diff
@@ -21,6 +21,12 @@ type Config struct {
 	// Config includes parameters that are the same in the source and destination.
 	common.Config
 
+	// StreamName is the name of the Kinesis Data Stream. It can contain a
+	// [Go template](https://pkg.go.dev/text/template) that will be executed
+	// for each record to determine the stream name. By default, the stream
+	// name is the value of the `opencdc.collection` metadata field.
+	StreamName string `json:"streamName"`
```

Comment:
```suggestion
	StreamName string `json:"streamName" default:"{{ index .Metadata \"opencdc.collection\" }}"`
```

Reply:
I'm not sure about this one. If the standard is to get the collection from the `opencdc.collection` record field, why would we put the overhead of a go template execution? Specially since the `.Write()` method is in the hot path.
If we change zerolog calls from `.Info()` to `.Trace()`, I believe we should also not use a go template when getting the collection from `opencdc.collection`.

If we however want to still preserve the readability of having a `default:"{{ index .Metadata \"opencdc.collection\"  }}` in a struct tag, we could also just hardcode that template, like this:

```go
func parseStreamName(config Config, rec sdk.Record) (string, error) {
	if config.StreamName == "{{ index .Metadata \"opencdc.collection\" }}" {
		return rec.Metadata.GetCollection()
	}

	// parse and execute template
}
```